### ISSUE TITLE
I2C Library

### ISSUE THREAD

**Issue by luca-byte** [Team-Isaac-Polito/MotorsBoard#5]
Define an Arduino library to deal with the I2C connection to the Main Board

- [ ] Identify the data that the motors can receive from the main-board [here](https://github.com/Team-Isaac-Polito/PicoLowLevel/blob/bfe21c8aba421f755014b434a067ea342692a642/PicoLowLevel.ino#L228-L229)
- [ ] Identify the data that should be sent back to the main-board [here](https://github.com/Team-Isaac-Polito/PicoLowLevel/blob/bfe21c8aba421f755014b434a067ea342692a642/PicoLowLevel.ino#L61-L62)
- [ ] Define some functions to setup the I2C channel, to receive data, to send data.

The I2C master should send data to the motors board and request telemetry after a certain time has passed, so this is the main use case. The operations of calibration, update, reset without too much time without instructions will be completely carried out by the motors board

For more info on the I2C protocol and its Arduino implementation: [Arduino Docs](https://docs.arduino.cc/learn/communication/wire/), [Pico Docs](https://arduino-pico.readthedocs.io/en/latest/wire.html#)

**Comment by Disteroas** [comment]
> Define an Arduino library to deal with the I2C connection to the Main Board
> 
> * [ ]  Identify the data that the motors can receive from the main-board [here](https://github.com/Team-Isaac-Polito/PicoLowLevel/blob/bfe21c8aba421f755014b434a067ea342692a642/PicoLowLevel.ino#L228-L229)
> * [ ]  Identify the data that should be sent back to the main-board [here](https://github.com/Team-Isaac-Polito/PicoLowLevel/blob/bfe21c8aba421f755014b434a067ea342692a642/PicoLowLevel.ino#L61-L62)
> * [ ]  Define some functions to setup the I2C channel, to receive data, to send data.
> 
> The I2C master should send data to the motors board and request telemetry after a certain time has passed, so this is the main use case. The operations of calibration, update, reset without too much time without instructions will be completely carried out by the motors board
> 
> For more info on the I2C protocol and its Arduino implementation: [Arduino Docs](https://docs.arduino.cc/learn/communication/wire/), [Pico Docs](https://arduino-pico.readthedocs.io/en/latest/wire.html#)

Tysm

**Comment by luca-byte** [comment]
This #2 should not be a problem until we want to use the temperature sensor

**Comment by Disteroas** [comment]
> This #2 should not be a problem until we want to use the temperature sensor

It is because we need an ADC?

**Comment by luca-byte** [comment]
> > This #2 should not be a problem until we want to use the temperature sensor
> 
> It is because we need an ADC?

No, both the sensor and the main board are connected to the same channel, so they interfere as far as I understood it. But for now I think that we can focus only on board-to-board communication and move this problem further away

**Comment by Disteroas** [comment]
> > > This #2 should not be a problem until we want to use the temperature sensor
> > 
> > 
> > It is because we need an ADC?
> 
> No, both the sensor and the main board are connected to the same channel, so they interfere as far as I understood it. But for now I think that we can focus only on board-to-board communication and move this problem further away

Ok perfect! (A great task for new members...)
